src/idpi/grib_decoder.py

```diff
@@ -91,227 +94,303 @@ def _extract_pv(pv):
     }
 
 
-def _update_origin(metadata, ref_param):
-    x0_key = "longitudeOfFirstGridPointInDegrees"
-    y0_key = "latitudeOfFirstGridPointInDegrees"
-    ref_geo = metadata[ref_param]["geography"]
-    x0 = ref_geo[x0_key] % 360
-    y0 = ref_geo[y0_key]
-    for m in metadata.values():
-        geo = m["geography"]
+@dc.dataclass
+class Grid:
+    lon: xr.DataArray
+    lat: xr.DataArray
+    lon_first_grid_point: float
+    lat_first_grid_point: float
```

Comment:
Maybe there should be some hint that the 2 first attributes are in the geographical CRS and the others are in the rotated CRS.

Reply:
done, I add the doc in the style of a method, not sure if we have some other pattern to doc dataclasses

Reply:
https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes